.github/workflows/test.yaml

```diff
@@ -107,4 +107,5 @@ jobs:
         path: |
           ~/go/src/github.com/DataDog/datadog-agent/test_output.json
           ~/go/src/github.com/DataDog/datadog-agent/junit-tests_macos.tgz
+          ~/go/src/github.com/DataDog/datadog-agent/junit-out-base.xml
```

Comment:
This only retrieves the junit file for the base module.

Each module has its own file in their module path (for instance, there is one in `pkg/trace/junit-out-base.xml`), it'd be nice to get all of them.

One complication here is that all these files have the same name, so you can't group them in the same directory without transformations.

To give an example, the `test` invoke task handles this when producing the `junit-tests_macos.tgz` file by prefixing each junit file with the path it was taken from.